# Chargement et Exploration des données

In [30]:
import pandas as pd

Affichons le contenu de nos datasets

In [31]:
data = pd.read_csv("../data/business.csv")
data.head()

date    airline ch_code  num_code dep_time   from time_taken  \
0  11-02-2022  Air India      AI       868    18:00  Delhi    02h 00m   
1  11-02-2022  Air India      AI       624    19:00  Delhi    02h 15m   
2  11-02-2022  Air India      AI       531    20:00  Delhi    24h 45m   
3  11-02-2022  Air India      AI       839    21:25  Delhi    26h 30m   
4  11-02-2022  Air India      AI       544    17:15  Delhi    06h 40m   

                                                stop arr_time      to   price  
0                                          non-stop     20:00  Mumbai  25,612  
1                                          non-stop     21:15  Mumbai  25,612  
2  1-stop\n\t\t\t\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t...    20:45  Mumbai  42,220  
3  1-stop\n\t\t\t\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t...    23:55  Mumbai  44,450  
4  1-stop\n\t\t\t\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t...    23:55  Mumbai  46,690

In [32]:
data = pd.read_csv("../data/economy.csv")
data.head()

date   airline ch_code  num_code dep_time   from time_taken  \
0  11-02-2022  SpiceJet      SG      8709    18:55  Delhi    02h 10m   
1  11-02-2022  SpiceJet      SG      8157    06:20  Delhi    02h 20m   
2  11-02-2022   AirAsia      I5       764    04:25  Delhi    02h 10m   
3  11-02-2022   Vistara      UK       995    10:20  Delhi    02h 15m   
4  11-02-2022   Vistara      UK       963    08:50  Delhi    02h 20m   

        stop arr_time      to  price  
0  non-stop     21:05  Mumbai  5,953  
1  non-stop     08:40  Mumbai  5,953  
2  non-stop     06:35  Mumbai  5,956  
3  non-stop     12:35  Mumbai  5,955  
4  non-stop     11:10  Mumbai  5,955

Nos deux DataFrames ont la même structure, nous allons par conséquent créer une nouvelle colonne pour la classe afin de spécifier s'il s'agit d'un vol en classe économique ou en classe business. Ensuite il faudra faire une concaténation des deux DataFrames

In [33]:
df1 = pd.read_csv("../data/business.csv")
df1["class"] = "business"
df2 = pd.read_csv("../data/economy.csv")
df2["class"] = "economy"

data = pd.concat([df1, df2], ignore_index=True)
data.head()

date    airline ch_code  num_code dep_time   from time_taken  \
0  11-02-2022  Air India      AI       868    18:00  Delhi    02h 00m   
1  11-02-2022  Air India      AI       624    19:00  Delhi    02h 15m   
2  11-02-2022  Air India      AI       531    20:00  Delhi    24h 45m   
3  11-02-2022  Air India      AI       839    21:25  Delhi    26h 30m   
4  11-02-2022  Air India      AI       544    17:15  Delhi    06h 40m   

                                                stop arr_time      to   price  \
0                                          non-stop     20:00  Mumbai  25,612   
1                                          non-stop     21:15  Mumbai  25,612   
2  1-stop\n\t\t\t\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t...    20:45  Mumbai  42,220   
3  1-stop\n\t\t\t\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t...    23:55  Mumbai  44,450   
4  1-stop\n\t\t\t\t\t\t\t\t\t\t\t\t\n\t\t\t\t\t\t...    23:55  Mumbai  46,690   

      class  
0  business  
1  business  
2  business  
3  business  
4  business

In [34]:
data.shape

(300261, 12)

Nous avons 300261 individus dans notre échantillon et 12 variables.

In [35]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300261 entries, 0 to 300260
Data columns (total 12 columns):
 #   Column      Non-Null Count   Dtype 
---  ------      --------------   ----- 
 0   date        300261 non-null  object
 1   airline     300261 non-null  object
 2   ch_code     300261 non-null  object
 3   num_code    300261 non-null  int64 
 4   dep_time    300261 non-null  object
 5   from        300261 non-null  object
 6   time_taken  300261 non-null  object
 7   stop        300261 non-null  object
 8   arr_time    300261 non-null  object
 9   to          300261 non-null  object
 10  price       300261 non-null  object
 11  class       300261 non-null  object
dtypes: int64(1), object(11)
memory usage: 27.5+ MB


- Déja, il n'y a aucune valeur manquante, ce qui est déja très bien 😉
- Il faudra convertir la colonne ```date``` en objet datetime
- les variables ```ch_code``` et ```stop``` nous serons inutile, donc à supprimer
- On transformera les heures de départ et d'arrivée des vols en variable catégorielle pour notre analyse
- On transformera également la colonne ```time_taken``` en objet datetime, puis en variable catégorielle
- On convertira les valeurs de la colonne ```price``` en valeurs numérique

In [36]:
data.duplicated().sum()

np.int64(2)

On trouve deux doublons, À supprimer !

### Sauvegarde des données concaténées

In [37]:
data.to_csv("../data/data_concatenated.csv", index=False)